### ISSUE TITLE
Model is not getting trained properly

### ISSUE THREAD

**Issue by sujit420** [cerndb/dist-keras#67]
I am trying to integrate standard algos of Collaborative filtering e.g. Matix Factorization(MF),NNMF,NMF etc. using dist-keras.
Everything runs successfully, but all prediction output is 0.
On the other hand if i run same code separately only using keras, i get all predictions fine.
1. trainer = ADAG(keras_model=model, worker_optimizer=self.optimizer, loss=self.loss,
                       num_workers=self.num_workers, batch_size=16, communication_window=5,num_epoch=100,
                       features_col=['userId','movieId'], label_col="rating")

trained_model = trainer.train(dataset)
This runs successfully but doesnt give any prediction other than 0 value
2. model.fit(x=[train.userId,train.movieId],y=train.rating,batch_size=64,epochs=5)
This runs as usual with all correct predictions.

Can anyone please guide me what I am doing wrong here?
